lldb/tools/lldb-dap/JSONUtils.h

```diff
@@ -250,14 +252,16 @@ protocol::Source CreateSource(const lldb::SBFileSpec &file);
 
 /// Create a "Source" JSON object as described in the debug adapter definition.
 ///
-/// \param[in] line_entry
-///     The LLDB line table to use when populating out the "Source"
-///     object
+/// \param[in] address
+///     The address to use when populating out the "Source" object.
+///
+/// \param[in] target
+///     The target that has the address.
 ///
 /// \return
 ///     A "Source" JSON object that follows the formal JSON
 ///     definition outlined by Microsoft.
-protocol::Source CreateSource(const lldb::SBLineEntry &line_entry);
+protocol::Source CreateSource(lldb::SBAddress address, lldb::SBTarget &target);
```

Comment:
Do we need to pass in the `SBTarget`? Isn't that set if we construct the `SBAddress` using the  https://github.com/llvm/llvm-project/blob/892bfa93662329314573cf0a8f5160a9d5d9b34d/lldb/include/lldb/API/SBAddress.h#L25C1-L27C1 constructor?

Reply:
the `SBTarget` is not saved there, it's only used in the constructor to parse the load address